### PR TITLE
feat: added types for the Accordion, AccordionItem, AccordionSkeleton, SkeletonText Components

### LINKED ISSUE #13544: Add TypeScript types to Accordion, AccordionItem, AccordionSkeleton

### PR DESCRIPTION
**Pull request by SunnyJohal**
Closes #13544 #13576 

Added type definitions to the Accordion, AccordionItem, AccordionSkeleton, SkeletonTextComponents

#### Changelog

**New**

- Now using context instead of cloning the children within the accordion and accordion item.

**Changed**

- Added type definitions to the Accordion, AccordionItem, AccordionSkeleton, SkeletonTextComponents.

**Removed**

- Unused IconDescription prop. 

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

### CODE REVIEW

**Comment by SunnyJohal** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/Accordion/Accordion.Skeleton.tsx [hunk 1]
```diff
@@ -12,14 +12,43 @@ import { ChevronRight } from '@carbon/icons-react';
...
+
 function AccordionSkeleton({
-  align,
+  align = 'end',
   className,
-  count,
+  count = 4,
   isFlush,
-  open,
+  open = true,
```
Comment (tay1orjones) [inline packages/react/src/components/Accordion/Accordion.Skeleton.tsx:49]: Thanks for adding these! You can remove `defaultProps`

File: packages/react/src/components/Accordion/Accordion.Skeleton.tsx [hunk 2]
```diff
@@ -12,14 +12,43 @@ import { ChevronRight } from '@carbon/icons-react';
 import SkeletonText from '../SkeletonText';
 import { usePrefix } from '../../internal/usePrefix';
 
+interface AccordionSkeletonProps {
+  /**
+   * @description Specify the alignment of the accordion heading
```
Comment (tay1orjones) [inline packages/react/src/components/Accordion/Accordion.Skeleton.tsx]: The `@description` JSDoc annotation is [not supported](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#unsupported-tags)

```suggestion
   * Specify the alignment of the accordion heading
```

Additional usages of `@description` in this PR will need updated as well

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Looks great, just a couple minor things

File: packages/react/src/components/Accordion/Accordion.Skeleton.tsx (hunk 2 above)
Comment (SunnyJohal) [reply packages/react/src/components/Accordion/Accordion.Skeleton.tsx]: Sure, I'll remove these now

File: packages/react/src/components/Accordion/Accordion.Skeleton.tsx (hunk 1 above)
Comment (SunnyJohal) [reply packages/react/src/components/Accordion/Accordion.Skeleton.tsx:49]: Great, will remove them now

**Comment by SunnyJohal** [comment]
@tay1orjones thanks for the feedback, I've made the changes 👍

**Review by andreancardona — APPROVED** [review]

**Comment by SunnyJohal** [comment]
@tay1orjones this branch appears to have a conflict, what's the best way forward to get this PR over the line?

**Comment by andreancardona** [comment]
@SunnyJohal for some reason the DCO is failing - I'll keep an eye on it but I have fixed the merge conflicts

**Comment by SunnyJohal** [comment]
Ahh, I think this is because I made a commit in this PR using a work e-mail address that I don't have access to anymore. I'll just close this and create a new branch so that we can get this merged in.

**Comment by SunnyJohal** [comment]
Closing this as I've created a [new PR](https://github.com/carbon-design-system/carbon/pull/13875) to get around the DCO issue.

**Pull request closed without merging**